### PR TITLE
CFY-3023-better-log-locations

### PR DESCRIPTION
**Pull request by geokala**
Ensures all manager logs end up in /var/log/cloudify

### CODE REVIEW

**Comment by nir0s** [comment]
Sandy, please resolve conflicts here. Thanks.

**Comment by m1keil** [comment]
LGTM

**Comment by geokala** [comment]
@nir0s: Rebased, it was the external elasticsearch change.

File: new/components/java/scripts/create.sh [hunk 1]
```diff
@@ -24,3 +24,10 @@ else
     ctx logger info "Installing Java..."
     sudo rpm -ivh /tmp/java.rpm >/dev/null
 fi
+
+# Make sure the cloudify logs dir exists before we try moving the java log there
+# -p will cause it not to error if the dir already exists
+sudo mkdir -p /var/log/cloudify
```
Comment (nir0s) [inline new/components/java/scripts/create.sh]: please use create_dir, here.

Comment (nir0s) [inline new/components/java/scripts/create.sh]: please use create_dir, here.

File: new/components/logstash/config/logstash [hunk 2]
```diff
@@ -0,0 +1,2 @@
+LS_LOG_DIR=/var/log/cloudify/logstash/
+LS_LOG_FILE=/var/log/cloudify/logstash/logstash.log
```
Comment (nir0s) [inline new/components/logstash/config/logstash]: this should probably be

```shell
LS_LOG_FILE=${LS_LOG_DIR}/logstash.log, right?

Comment (nir0s) [inline new/components/logstash/config/logstash]: forget it. sysconfig file.

File: new/components/utils [hunk 3]
```diff
@@ -385,6 +385,17 @@ function create_service_user() {
...
+function clean_var_log_dir() {
+    ###
+    # Cleans up unused /var/log directory for named application.
+    # Directory must be empty or this will fail.
+    ###
+    for log in $(find /var/log/${1} -type f); do
+        sudo mv ${log} /var/log/cloudify/${1}/${log##/var/log/${1}/}-from_bootstrap
+    done
+    ctx logger info "Removing unnecessary logs directory: /var/log/${1}"
+    sudo rmdir /var/log/${1}
```
Comment (nir0s) [inline new/components/utils]: is this idempotent?

Comment (geokala) [inline new/components/utils]: It's not, it'll fail if the dir doesn't exist (so on second execution).

I'll make a couple of changes and upload them shortly.

Comment (nir0s) [inline new/components/utils]: thanks

**Pull request merged**